include/zmq.h

```diff
@@ -637,6 +637,8 @@ ZMQ_EXPORT void *zmq_threadstart (zmq_thread_fn *func_, void *arg_);
 /* Wait for thread to complete then free up resources.                        */
 ZMQ_EXPORT void zmq_threadclose (void *thread_);
 
+/* Equivalent of snprintf that asserts on error or truncation.                */
+ZMQ_EXPORT int zmq_snprintf ( char *buffer, size_t bufsz, const char *format, ... );
```

Comment:
no new public APIs for this please, we do not want to be in the business of replacing glibc. There's just 2 cases using the return value, simply fix those instead

Reply:
Adding this function isn't equivalent to checking the return value only in cases where that value was used (because it always asserts on errors and truncation, as it should).

It wasn't clear to me what the convention was for adding an internal helper function; what header and source file should that use?

Reply:
There's no need for anything special for the tests, it really doesn't matter there. Outside of the tests there's only 2 cases, so please just do the required checks directly there

Reply:
Actually I'll take a different approach that uses `sprintf_s`; that doesn't require any helper function.